dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -138,189 +224,318 @@ def _submit_job(self, path: str, cluster_spec: dict) -> str:
+    @staticmethod
+    def cluster_settings(parsed_model: ParsedPythonModel) -> Dict[str, Any]:
+        config = parsed_model.config
+        job_cluster_config = config.job_cluster_config
 
-    def _build_job_spec(self) -> Dict[str, Any]:
-        workflow_spec = dict(self.workflow_spec)
-        workflow_spec["name"] = self.workflow_spec.get("name", self.default_job_name)
+        cluster_settings: Dict[str, Any] = {}
+        if job_cluster_config:
+            cluster_settings["new_cluster"] = job_cluster_config
+        elif config.cluster_id:
+            cluster_settings["existing_cluster_id"] = config.cluster_id
 
-        # Undefined cluster settings defaults to serverless in the Databricks API
-        cluster_settings = {}
-        if self.cluster_spec:
-            cluster_settings["new_cluster"] = self.cluster_spec
-        elif "existing_cluster_id" in self.workflow_spec:
-            cluster_settings["existing_cluster_id"] = self.workflow_spec["existing_cluster_id"]
+        return cluster_settings
 
+    def compile(self, path: str) -> Tuple[Dict[str, Any], str]:
         notebook_task = {
             "task_key": "inner_notebook",
             "notebook_task": {
-                "notebook_path": self.notebook_path,
+                "notebook_path": path,
                 "source": "WORKSPACE",
             },
         }
-        notebook_task.update(cluster_settings)
-        notebook_task.update(self.additional_task_settings)
+        notebook_task.update(self.task_settings)
+
+        self.workflow_spec["tasks"] = [notebook_task] + self.post_hook_tasks
+        return self.workflow_spec, self.existing_job_id
+
+
+class PythonWorkflowCreater:
+    """Manages the creation or updating of a workflow job on Databricks."""
+
+    def __init__(self, workflows: WorkflowJobApi) -> None:
+        self.workflows = workflows
 
-        workflow_spec["tasks"] = [notebook_task] + self.post_hook_tasks
-        return workflow_spec
+    def create_or_update(
+        self,
+        workflow_spec: Dict[str, Any],
+        existing_job_id: Optional[str],
+    ) -> str:
+        """
+        :return: tuple of job_id and whether the job is new
+        """
+        if not existing_job_id:
+            workflow_name = workflow_spec["name"]
+            response_jobs = self.workflows.search_by_name(workflow_name)
+
+            if len(response_jobs) > 1:
+                raise DbtRuntimeError(
+                    f"Multiple jobs found with name {workflow_name}. Use a"
+                    " unique job name or specify the `existing_job_id` in the python_job_config."
+                )
 
-    def _submit_through_workflow(self, compiled_code: str, workflow_spec: Dict[str, Any]) -> None:
-        self.api_client.workspace.create_python_model_dir(self.catalog, self.schema)
-        self.api_client.workspace.upload_notebook(self.notebook_path, compiled_code)
+            if len(response_jobs) == 1:
```

Comment:
nit: `elif`

Reply:
so, elif is not needed, but I guess as a matter of style, it makes sense since it's mutually exclusive with the condition above.